pkg/storage/gcp/bucket.go

```diff
@@ -0,0 +1,21 @@
+package gcp
+
+import (
+	"context"
+	"io"
+)
+
+// Bucket defines the storage bucket TODO
+type Bucket interface {
```

Comment:
I'm a tiny bit confused about this interface. Some methods seem to work directly with a whole module (DeleteModule, ListVersions) but some are about files/objects within modules? If that's true that maybe a little documentation about why that is would be great. Otherwise, maybe having this interface not know anything about modules and just work with GCP terminology, or the other way around.

Reply:
I'll fix it. It was because I didn't want the extra code to convert module, version into the file path. It will just require the signature to include the file extension.

Reply:
So to clarify, it does not operate on a file within the module. but the module at the path filename.